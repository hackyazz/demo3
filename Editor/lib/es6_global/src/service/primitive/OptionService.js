

import * as Curry from "../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as Caml_option from "../../../../../node_modules/bs-platform/lib/es6/caml_option.js";
import * as OptionService$Wonderjs from "../../../../../node_modules/wonder.js/lib/es6_global/src/service/atom/OptionService.js";

function join(x) {
  if (x !== undefined) {
    return Caml_option.valFromOption(x);
  }
  
}

function andThenWithDefault(func, $$default, x) {
  if (x !== undefined) {
    return Curry._1(func, Caml_option.valFromOption(x));
  } else {
    return $$default;
  }
}

function either(someFunc, noneFunc, data, x) {
  if (x !== undefined) {
    return Curry._2(someFunc, data, Caml_option.valFromOption(x));
  } else {
    return Curry._1(noneFunc, data);
  }
}

function eitherWithNoData(someFunc, noneFunc, x) {
  if (x !== undefined) {
    return Curry._1(someFunc, Caml_option.valFromOption(x));
  } else {
    return Curry._1(noneFunc, /* () */0);
  }
}

function handleSomeAndIgnore(func, x) {
  if (x !== undefined) {
    Curry._1(func, Caml_option.valFromOption(x));
    return /* () */0;
  } else {
    return /* () */0;
  }
}

var unsafeGet = OptionService$Wonderjs.unsafeGet;

var unsafeGetJsonSerializedValue = OptionService$Wonderjs.unsafeGetJsonSerializedValue;

var isJsonSerializedValueNone = OptionService$Wonderjs.isJsonSerializedValueNone;

export {
  unsafeGet ,
  unsafeGetJsonSerializedValue ,
  isJsonSerializedValueNone ,
  join ,
  andThenWithDefault ,
  either ,
  eitherWithNoData ,
  handleSomeAndIgnore ,
  
}
/* OptionService-Wonderjs Not a pure module */
