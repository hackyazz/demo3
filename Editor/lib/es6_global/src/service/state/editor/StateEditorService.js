

import * as CreateEditorStateDataEditorService$WonderEditor from "../data/CreateEditorStateDataEditorService.js";

function getStateIsDebug(param) {
  return CreateEditorStateDataEditorService$WonderEditor.editorStateData[/* isDebug */1];
}

function setStateIsDebug(isDebug) {
  CreateEditorStateDataEditorService$WonderEditor.editorStateData[/* isDebug */1] = isDebug;
  return /* () */0;
}

function getIsRun(param) {
  return CreateEditorStateDataEditorService$WonderEditor.editorStateData[/* isRun */2];
}

function setIsRun(isRun) {
  CreateEditorStateDataEditorService$WonderEditor.editorStateData[/* isRun */2] = isRun;
  return /* () */0;
}

function getState(param) {
  return CreateEditorStateDataEditorService$WonderEditor.editorStateData[/* editorState */0];
}

function setState(state) {
  CreateEditorStateDataEditorService$WonderEditor.editorStateData[/* editorState */0] = state;
  return state;
}

export {
  getStateIsDebug ,
  setStateIsDebug ,
  getIsRun ,
  setIsRun ,
  getState ,
  setState ,
  
}
/* CreateEditorStateDataEditorService-WonderEditor Not a pure module */
