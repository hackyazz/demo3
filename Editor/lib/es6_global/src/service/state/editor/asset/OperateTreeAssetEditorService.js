

import * as Curry from "../../../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as Js_option from "../../../../../../../node_modules/bs-platform/lib/es6/js_option.js";
import * as NodeAssetService$WonderEditor from "../../../record/editor/asset/NodeAssetService.js";
import * as FolderNodeAssetService$WonderEditor from "../../../record/editor/asset/FolderNodeAssetService.js";
import * as TreeAssetEditorService$WonderEditor from "./TreeAssetEditorService.js";
import * as IterateTreeAssetService$WonderEditor from "../../../record/editor/asset/IterateTreeAssetService.js";
import * as OperateTreeAssetService$WonderEditor from "../../../record/editor/asset/OperateTreeAssetService.js";
import * as MaterialNodeAssetService$WonderEditor from "../../../record/editor/asset/MaterialNodeAssetService.js";
import * as CurrentNodeIdAssetEditorService$WonderEditor from "./CurrentNodeIdAssetEditorService.js";

function _getTree(func, editorState) {
  return Curry._1(func, TreeAssetEditorService$WonderEditor.unsafeGetTree(editorState));
}

function _getTreeWithLabel(func, editorState) {
  return Curry._1(func, TreeAssetEditorService$WonderEditor.unsafeGetTree(editorState));
}

function _getAndSetTree(func, editorState) {
  var __x = Curry._1(func, TreeAssetEditorService$WonderEditor.unsafeGetTree(editorState));
  return TreeAssetEditorService$WonderEditor.setTree(__x, editorState);
}

function findNodeById(targetNodeId, editorState) {
  var param = TreeAssetEditorService$WonderEditor.unsafeGetTree(editorState);
  return OperateTreeAssetService$WonderEditor.findNodeById(targetNodeId, param);
}

function unsafeFindNodeById(targetNodeId, editorState) {
  var param = TreeAssetEditorService$WonderEditor.unsafeGetTree(editorState);
  return OperateTreeAssetService$WonderEditor.unsafeFindNodeById(targetNodeId, param);
}

function isNodeExistById(targetNodeId, editorState) {
  var param = TreeAssetEditorService$WonderEditor.unsafeGetTree(editorState);
  return Js_option.isSome(OperateTreeAssetService$WonderEditor.findNodeById(targetNodeId, param));
}

function findMaterialNode(targetMaterialComponent, targetType_, editorState) {
  var predNodeFunc = function (materialNode) {
    var match = MaterialNodeAssetService$WonderEditor.getNodeData(materialNode);
    if (match[/* materialComponent */1] === targetMaterialComponent) {
      return match[/* type_ */0] === targetType_;
    } else {
      return false;
    }
  };
  var arg = predNodeFunc;
  return (function (param) {
              return IterateTreeAssetService$WonderEditor.findOne(param, undefined, arg, undefined, undefined, /* () */0);
            })(TreeAssetEditorService$WonderEditor.unsafeGetTree(editorState));
}

function insertNode(targetNodeId, newTreeNode, editorState) {
  return _getAndSetTree((function (param) {
                return OperateTreeAssetService$WonderEditor.insertNode(targetNodeId, newTreeNode, param);
              }), editorState);
}

function setNodeIsShowChildren(targetNodeId, isShowChildren, editorState) {
  return _getAndSetTree((function (param) {
                return OperateTreeAssetService$WonderEditor.setNodeIsShowChildren(targetNodeId, isShowChildren, param);
              }), editorState);
}

function removeNode(targetNode, editorState) {
  var newTree = OperateTreeAssetService$WonderEditor.removeNode(targetNode, TreeAssetEditorService$WonderEditor.unsafeGetTree(editorState));
  return TreeAssetEditorService$WonderEditor.setTree(newTree, editorState);
}

function removeNodeById(targetNodeId, editorState) {
  var match = OperateTreeAssetService$WonderEditor.removeNodeById(targetNodeId, TreeAssetEditorService$WonderEditor.unsafeGetTree(editorState));
  var newTree = match[0];
  return /* tuple */[
          (function (param) {
              return TreeAssetEditorService$WonderEditor.setTree(newTree, param);
            }),
          match[1]
        ];
}

function findNodeParent(targetNode, editorState) {
  return IterateTreeAssetService$WonderEditor.findOne(TreeAssetEditorService$WonderEditor.unsafeGetTree(editorState), undefined, undefined, undefined, (function (node) {
                return Js_option.isSome(FolderNodeAssetService$WonderEditor.findChild(node, targetNode));
              }), /* () */0);
}

function findNodeParentId(targetNode, editorState) {
  return Js_option.map(NodeAssetService$WonderEditor.getNodeId, findNodeParent(targetNode, editorState));
}

function getCurrentNode(editorState) {
  return Js_option.map((function (nodeId) {
                var param = TreeAssetEditorService$WonderEditor.unsafeGetTree(editorState);
                return OperateTreeAssetService$WonderEditor.unsafeFindNodeById(nodeId, param);
              }), CurrentNodeIdAssetEditorService$WonderEditor.getCurrentNodeId(editorState));
}

function unsafeGetSelectedFolderNodeInAssetTree(editorState) {
  var targetNodeId = TreeAssetEditorService$WonderEditor.getSelectedFolderNodeIdInAssetTree(editorState);
  var param = TreeAssetEditorService$WonderEditor.unsafeGetTree(editorState);
  return OperateTreeAssetService$WonderEditor.unsafeFindNodeById(targetNodeId, param);
}

export {
  _getTree ,
  _getTreeWithLabel ,
  _getAndSetTree ,
  findNodeById ,
  unsafeFindNodeById ,
  isNodeExistById ,
  findMaterialNode ,
  insertNode ,
  setNodeIsShowChildren ,
  removeNode ,
  removeNodeById ,
  findNodeParent ,
  findNodeParentId ,
  getCurrentNode ,
  unsafeGetSelectedFolderNodeInAssetTree ,
  
}
/* FolderNodeAssetService-WonderEditor Not a pure module */
