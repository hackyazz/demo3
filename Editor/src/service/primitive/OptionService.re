open Wonderjs;

let unsafeGet = OptionService.unsafeGet;

let unsafeGetJsonSerializedValue = OptionService.unsafeGetJsonSerializedValue;

let isJsonSerializedValueNone = OptionService.isJsonSerializedValueNone;

let join = x =>
  switch (x) {
  | None => None
  | Some(x) => x
  };

let andThenWithDefault = (func, default, x) =>
  switch (x) {
  | None => default
  | Some(x) => func(x)
  };

let either = (someFunc, noneFunc, data, x) =>
  switch (x) {
  | None => noneFunc(data)
  | Some(x) => someFunc(data, x)
  };

let eitherWithNoData = (someFunc, noneFunc, x) =>
  switch (x) {
  | None => noneFunc()
  | Some(x) => someFunc(x)
  };

let handleSomeAndIgnore = (func, x) =>
  switch (x) {
  | None => ()
  | Some(x) => func(x) |> ignore
  };