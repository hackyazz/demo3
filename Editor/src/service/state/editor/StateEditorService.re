open EditorStateDataType;

let getStateIsDebug = () =>
  CreateEditorStateDataEditorService.editorStateData.isDebug;

let setStateIsDebug = isDebug => {
  CreateEditorStateDataEditorService.editorStateData.isDebug = isDebug;

  ();
};

let getIsRun = () => CreateEditorStateDataEditorService.editorStateData.isRun;

let setIsRun = isRun => {
  CreateEditorStateDataEditorService.editorStateData.isRun = isRun;

  ();
};

let getState = () =>
  CreateEditorStateDataEditorService.editorStateData.editorState;

let setState = state => {
  CreateEditorStateDataEditorService.editorStateData.editorState = state;
  state;
};