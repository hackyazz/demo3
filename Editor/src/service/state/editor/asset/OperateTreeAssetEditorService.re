open NodeAssetType;

let _getTree = (func, editorState) =>
  TreeAssetEditorService.unsafeGetTree(editorState) |> func;

let _getTreeWithLabel = (func, editorState) =>
  func(~tree=TreeAssetEditorService.unsafeGetTree(editorState));

let _getAndSetTree = (func, editorState) =>
  TreeAssetEditorService.unsafeGetTree(editorState)
  |> func
  |> TreeAssetEditorService.setTree(_, editorState);

let findNodeById = (targetNodeId, editorState) =>
  _getTree(OperateTreeAssetService.findNodeById(targetNodeId), editorState);

let unsafeFindNodeById = (targetNodeId, editorState) =>
  _getTree(
    OperateTreeAssetService.unsafeFindNodeById(targetNodeId),
    editorState,
  );

let isNodeExistById = (targetNodeId, editorState) =>
  findNodeById(targetNodeId, editorState) |> Js.Option.isSome;

let findMaterialNode = (targetMaterialComponent, targetType_, editorState) => {
  let predNodeFunc = materialNode => {
    let {materialComponent, type_} =
      MaterialNodeAssetService.getNodeData(materialNode);

    materialComponent === targetMaterialComponent && type_ === targetType_;
  };

  editorState
  |> _getTreeWithLabel(
       IterateTreeAssetService.findOne(
         ~predMaterialNodeFunc=predNodeFunc,
         (),
       ),
     );
};

let insertNode = (targetNodeId, newTreeNode, editorState) =>
  _getAndSetTree(
    OperateTreeAssetService.insertNode(targetNodeId, newTreeNode),
    editorState,
  );

let setNodeIsShowChildren = (targetNodeId, isShowChildren, editorState) =>
  _getAndSetTree(
    OperateTreeAssetService.setNodeIsShowChildren(
      targetNodeId,
      isShowChildren,
    ),
    editorState,
  );

let removeNode = (targetNode, editorState) => {
  let newTree =
    TreeAssetEditorService.unsafeGetTree(editorState)
    |> OperateTreeAssetService.removeNode(targetNode);

  TreeAssetEditorService.setTree(newTree, editorState);
};

let removeNodeById = (targetNodeId, editorState) => {
  let (newTree, removedNode) =
    TreeAssetEditorService.unsafeGetTree(editorState)
    |> OperateTreeAssetService.removeNodeById(targetNodeId);

  (newTree |> TreeAssetEditorService.setTree, removedNode);
};

let findNodeParent = (targetNode, editorState) =>
  IterateTreeAssetService.findOne(
    ~tree=TreeAssetEditorService.unsafeGetTree(editorState),
    ~predFolderNodeFunc=
      node =>
        FolderNodeAssetService.findChild(node, targetNode) |> Js.Option.isSome,
    (),
  );

let findNodeParentId = (targetNode, editorState) =>
  findNodeParent(targetNode, editorState)
  |> Js.Option.map((. node) => NodeAssetService.getNodeId(~node));

let getCurrentNode = editorState =>
  CurrentNodeIdAssetEditorService.getCurrentNodeId(editorState)
  |> Js.Option.map((. nodeId) => unsafeFindNodeById(nodeId, editorState));

let unsafeGetSelectedFolderNodeInAssetTree = editorState =>
  TreeAssetEditorService.getSelectedFolderNodeIdInAssetTree(editorState)
  |> unsafeFindNodeById(_, editorState);