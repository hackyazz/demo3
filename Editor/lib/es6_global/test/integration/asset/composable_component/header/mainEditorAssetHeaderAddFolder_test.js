

import * as Curry from "../../../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as Sinon from "../../../../../../../node_modules/wonder-bs-sinon/lib/es6_global/src/sinon.js";
import * as Sinon$1 from "sinon";
import * as Wonder_jest from "../../../../../../../node_modules/wonder-bs-jest/lib/es6_global/src/wonder_jest.js";
import * as OptionService$WonderEditor from "../../../../../src/service/primitive/OptionService.js";
import * as ReactTestTool$WonderEditor from "../../../../tool/ReactTestTool.js";
import * as EventListenerTool$WonderEditor from "../../../../unit/tool/EventListenerTool.js";
import * as BuildComponentTool$WonderEditor from "../../../../tool/BuildComponentTool.js";
import * as StateEditorService$WonderEditor from "../../../../../src/service/state/editor/StateEditorService.js";
import * as MainEditorSceneTool$WonderEditor from "../../../../tool/MainEditorSceneTool.js";
import * as MainEditorAssetIdTool$WonderEditor from "../../tool/MainEditorAssetIdTool.js";
import * as MainEditorAssetTreeTool$WonderEditor from "../../tool/MainEditorAssetTreeTool.js";
import * as OperateTreeAssetEditorService$WonderEditor from "../../../../../src/service/state/editor/asset/OperateTreeAssetEditorService.js";
import * as MainEditorAssetHeaderOperateNodeTool$WonderEditor from "../../tool/MainEditorAssetHeaderOperateNodeTool.js";

describe("MainEditorAssetHeader->add folder", (function () {
        var sandbox = Sinon.getSandboxDefaultVal(/* () */0);
        beforeEach((function () {
                sandbox[0] = Sinon$1.sandbox.create();
                MainEditorSceneTool$WonderEditor.initState(sandbox, undefined, undefined, /* () */0);
                MainEditorSceneTool$WonderEditor.createDefaultScene(sandbox, (function () {
                        return /* () */0;
                      }));
                return Curry._1(EventListenerTool$WonderEditor.stubGetElementByIdReturnFakeDom, Curry._1(EventListenerTool$WonderEditor.buildFakeDom, /* () */0));
              }));
        afterEach((function () {
                return Curry._1(Sinon.restoreSandbox, sandbox[0]);
              }));
        describe("if not select specific treeNode, add folder into root treeNode", (function () {
                describe("should add folder into root treeNode", (function () {
                        Wonder_jest.test("test snapshot", (function () {
                                Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Folder */5][/* TwoLayer */0][/* buildOneFolderAssetTree */0], /* () */0);
                                MainEditorAssetHeaderOperateNodeTool$WonderEditor.addFolder(undefined, undefined, /* () */0);
                                return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildAssetComponent(/* () */0));
                              }));
                        return Wonder_jest.test("the added folder parent node should be root", (function () {
                                      var assetTreeData = Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Folder */5][/* TwoLayer */0][/* buildOneFolderAssetTree */0], /* () */0);
                                      var addedFolderNodeId = MainEditorAssetIdTool$WonderEditor.getNewAssetId(undefined, /* () */0);
                                      MainEditorAssetHeaderOperateNodeTool$WonderEditor.addFolder(undefined, undefined, /* () */0);
                                      var editorState = StateEditorService$WonderEditor.getState(/* () */0);
                                      return Curry._2(Wonder_jest.Expect[/* Operators */23][/* = */5], Wonder_jest.Expect[/* expect */0](OptionService$WonderEditor.unsafeGet(MainEditorAssetTreeTool$WonderEditor.findNodeParentId(OperateTreeAssetEditorService$WonderEditor.unsafeFindNodeById(addedFolderNodeId, editorState), editorState))), Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Folder */5][/* TwoLayer */0][/* getRootNodeId */2], assetTreeData));
                                    }));
                      }));
                return /* () */0;
              }));
        describe("else", (function () {
                describe("add folder into specific treeNode", (function () {
                        return Wonder_jest.test("test snapshot", (function () {
                                      var assetTreeData = Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Folder */5][/* TwoLayer */0][/* buildOneFolderAssetTree */0], /* () */0);
                                      MainEditorAssetTreeTool$WonderEditor.Select[/* selectFolderNode */1](Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Folder */5][/* TwoLayer */0][/* getFirstFolderNodeId */3], assetTreeData), undefined, /* () */0);
                                      MainEditorAssetHeaderOperateNodeTool$WonderEditor.addFolder(undefined, undefined, /* () */0);
                                      return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildAssetTree(/* () */0));
                                    }));
                      }));
                return /* () */0;
              }));
        describe("test name", (function () {
                Wonder_jest.test("test add the same name folder, the name should add postfix", (function () {
                        Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Folder */5][/* TwoLayer */0][/* buildOneFolderAssetTree */0], /* () */0);
                        MainEditorAssetHeaderOperateNodeTool$WonderEditor.addFolder(undefined, undefined, /* () */0);
                        MainEditorAssetHeaderOperateNodeTool$WonderEditor.addFolder(undefined, undefined, /* () */0);
                        MainEditorAssetHeaderOperateNodeTool$WonderEditor.addFolder(undefined, undefined, /* () */0);
                        return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildAssetTree(/* () */0));
                      }));
                Wonder_jest.test("test select folder and add the same name folder, the name should add postfix", (function () {
                        var assetTreeData = Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Folder */5][/* TwoLayer */0][/* buildOneFolderAssetTree */0], /* () */0);
                        var rootNodeId = Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Folder */5][/* TwoLayer */0][/* getRootNodeId */2], assetTreeData);
                        MainEditorAssetTreeTool$WonderEditor.Select[/* selectFolderNode */1](rootNodeId, undefined, /* () */0);
                        MainEditorAssetHeaderOperateNodeTool$WonderEditor.addFolder(undefined, undefined, /* () */0);
                        MainEditorAssetHeaderOperateNodeTool$WonderEditor.addFolder(undefined, undefined, /* () */0);
                        return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildAssetTree(/* () */0));
                      }));
                return Wonder_jest.test("remove first folder which use default name;\n          add three folder;\n\n          the first new one's name should be removed-folder's name;\n            ", (function () {
                              var assetTreeData = Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Folder */5][/* TwoLayer */0][/* buildOneFolderAssetTree */0], /* () */0);
                              MainEditorAssetHeaderOperateNodeTool$WonderEditor.removeFolderNode(undefined, undefined, Curry._1(MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* Folder */5][/* TwoLayer */0][/* getFirstFolderNodeId */3], assetTreeData), /* () */0);
                              MainEditorAssetHeaderOperateNodeTool$WonderEditor.addFolder(undefined, undefined, /* () */0);
                              MainEditorAssetHeaderOperateNodeTool$WonderEditor.addFolder(undefined, undefined, /* () */0);
                              MainEditorAssetHeaderOperateNodeTool$WonderEditor.addFolder(undefined, undefined, /* () */0);
                              return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildAssetTree(/* () */0));
                            }));
              }));
        return /* () */0;
      }));

export {
  
}
/*  Not a pure module */
