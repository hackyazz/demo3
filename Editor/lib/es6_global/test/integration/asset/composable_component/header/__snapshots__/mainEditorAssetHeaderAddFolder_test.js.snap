// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainEditorAssetHeader->add folder else add folder into specific treeNode test snapshot 1`] = `
<article
  className="wonder-asset-assetTree"
>
  <ul
    className="wonder-tree-node"
  >
    <li>
      <div
        className="item-triangle"
        onMouseDown={[Function]}
      >
        <img
          src="./public/img/down.png"
        />
      </div>
      <img
        className="treeNode-icon"
        src="./public/img/package.png"
      />
      <div
        className="draggable-container"
        draggable={true}
        onClick={[Function]}
        onDragEnd={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        style={Object {}}
      >
        Assets
      </div>
    </li>
    <ul
      className="wonder-tree-node"
    >
      <li>
        <div
          className="item-triangle"
          onMouseDown={[Function]}
        >
          <img
            src="./public/img/down.png"
          />
        </div>
        <img
          className="treeNode-icon"
          src="./public/img/package.png"
        />
        <div
          className="draggable-container select-active"
          draggable={true}
          onClick={[Function]}
          onDragEnd={[Function]}
          onDragEnter={[Function]}
          onDragLeave={[Function]}
          onDragOver={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          style={Object {}}
        >
          NoName Folder
        </div>
      </li>
      <ul
        className="wonder-tree-node"
      >
        <li>
          <div
            className="item-triangle"
          />
          <img
            className="treeNode-icon"
            src="./public/img/package.png"
          />
          <div
            className="draggable-container"
            draggable={true}
            onClick={[Function]}
            onDragEnd={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            style={Object {}}
          >
            New Folder
          </div>
        </li>
      </ul>
    </ul>
  </ul>
</article>
`;

exports[`MainEditorAssetHeader->add folder if not select specific treeNode, add folder into root treeNode should add folder into root treeNode test snapshot 1`] = `
<article
  className="wonder-asset-component"
>
  <div
    className="asset-tree"
  >
    <article
      className="wonder-asset-header"
    >
      <div
        className="asset-header-item"
        onClick={[Function]}
      >
        <div
          className="item-canBeClick"
        >
          <img
            src="./public/img/add.png"
          />
        </div>
      </div>
      <div
        className="asset-header-item"
        onClick={[Function]}
      >
        <div
          className="item-notBeClick"
        >
          <img
            src="./public/img/notRemove.png"
          />
        </div>
      </div>
      <div
        className="asset-header-item"
      >
        <div
          className="item-canBeClick"
        >
          <img
            src="./public/img/load.png"
          />
          <input
            className="asset-fileLoad"
            multiple={false}
            onChange={[Function]}
            type="file"
          />
        </div>
      </div>
    </article>
    <article
      className="wonder-asset-assetTree"
    >
      <ul
        className="wonder-tree-node"
      >
        <li>
          <div
            className="item-triangle"
            onMouseDown={[Function]}
          >
            <img
              src="./public/img/down.png"
            />
          </div>
          <img
            className="treeNode-icon"
            src="./public/img/package.png"
          />
          <div
            className="draggable-container select-not-active"
            draggable={true}
            onClick={[Function]}
            onDragEnd={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            style={Object {}}
          >
            Assets
          </div>
        </li>
        <ul
          className="wonder-tree-node"
        >
          <li>
            <div
              className="item-triangle"
            />
            <img
              className="treeNode-icon"
              src="./public/img/package.png"
            />
            <div
              className="draggable-container"
              draggable={true}
              onClick={[Function]}
              onDragEnd={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDragStart={[Function]}
              onDrop={[Function]}
              style={Object {}}
            >
              NoName Folder
            </div>
          </li>
        </ul>
        <ul
          className="wonder-tree-node"
        >
          <li>
            <div
              className="item-triangle"
            />
            <img
              className="treeNode-icon"
              src="./public/img/package.png"
            />
            <div
              className="draggable-container"
              draggable={true}
              onClick={[Function]}
              onDragEnd={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDragStart={[Function]}
              onDrop={[Function]}
              style={Object {}}
            >
              New Folder
            </div>
          </li>
        </ul>
      </ul>
    </article>
  </div>
  <article
    className="wonder-asset-assetChildren"
  >
    <article
      className="wonder-asset-folderBox"
      id="folder-3"
      style={
        Object {
          "border": "0px",
        }
      }
    >
      <div
        className="item-ground"
        draggable={true}
        onDragEnd={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
      />
      <div
        className="box-image"
      >
        <img
          src="./public/img/assetPackage.png"
        />
      </div>
      <div
        className="item-text "
      >
        <span>
          NoName Folder
        </span>
      </div>
    </article>
    <article
      className="wonder-asset-folderBox"
      id="folder-4"
      style={
        Object {
          "border": "0px",
        }
      }
    >
      <div
        className="item-ground"
        draggable={true}
        onDragEnd={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
      />
      <div
        className="box-image"
      >
        <img
          src="./public/img/assetPackage.png"
        />
      </div>
      <div
        className="item-text "
      >
        <span>
          New Folder
        </span>
      </div>
    </article>
  </article>
</article>
`;

exports[`MainEditorAssetHeader->add folder test name remove first folder which use default name;
          add three folder;

          the first new one's name should be removed-folder's name;
             1`] = `
<article
  className="wonder-asset-assetTree"
>
  <ul
    className="wonder-tree-node"
  >
    <li>
      <div
        className="item-triangle"
        onMouseDown={[Function]}
      >
        <img
          src="./public/img/down.png"
        />
      </div>
      <img
        className="treeNode-icon"
        src="./public/img/package.png"
      />
      <div
        className="draggable-container select-not-active"
        draggable={true}
        onClick={[Function]}
        onDragEnd={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        style={Object {}}
      >
        Assets
      </div>
    </li>
    <ul
      className="wonder-tree-node"
    >
      <li>
        <div
          className="item-triangle"
        />
        <img
          className="treeNode-icon"
          src="./public/img/package.png"
        />
        <div
          className="draggable-container"
          draggable={true}
          onClick={[Function]}
          onDragEnd={[Function]}
          onDragEnter={[Function]}
          onDragLeave={[Function]}
          onDragOver={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          style={Object {}}
        >
          New Folder
        </div>
      </li>
    </ul>
    <ul
      className="wonder-tree-node"
    >
      <li>
        <div
          className="item-triangle"
        />
        <img
          className="treeNode-icon"
          src="./public/img/package.png"
        />
        <div
          className="draggable-container"
          draggable={true}
          onClick={[Function]}
          onDragEnd={[Function]}
          onDragEnter={[Function]}
          onDragLeave={[Function]}
          onDragOver={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          style={Object {}}
        >
          New Folder 1
        </div>
      </li>
    </ul>
    <ul
      className="wonder-tree-node"
    >
      <li>
        <div
          className="item-triangle"
        />
        <img
          className="treeNode-icon"
          src="./public/img/package.png"
        />
        <div
          className="draggable-container"
          draggable={true}
          onClick={[Function]}
          onDragEnd={[Function]}
          onDragEnter={[Function]}
          onDragLeave={[Function]}
          onDragOver={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          style={Object {}}
        >
          New Folder 2
        </div>
      </li>
    </ul>
  </ul>
</article>
`;

exports[`MainEditorAssetHeader->add folder test name test select folder and add the same name folder, the name should add postfix 1`] = `
<article
  className="wonder-asset-assetTree"
>
  <ul
    className="wonder-tree-node"
  >
    <li>
      <div
        className="item-triangle"
        onMouseDown={[Function]}
      >
        <img
          src="./public/img/down.png"
        />
      </div>
      <img
        className="treeNode-icon"
        src="./public/img/package.png"
      />
      <div
        className="draggable-container select-active"
        draggable={true}
        onClick={[Function]}
        onDragEnd={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        style={Object {}}
      >
        Assets
      </div>
    </li>
    <ul
      className="wonder-tree-node"
    >
      <li>
        <div
          className="item-triangle"
        />
        <img
          className="treeNode-icon"
          src="./public/img/package.png"
        />
        <div
          className="draggable-container"
          draggable={true}
          onClick={[Function]}
          onDragEnd={[Function]}
          onDragEnter={[Function]}
          onDragLeave={[Function]}
          onDragOver={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          style={Object {}}
        >
          NoName Folder
        </div>
      </li>
    </ul>
    <ul
      className="wonder-tree-node"
    >
      <li>
        <div
          className="item-triangle"
        />
        <img
          className="treeNode-icon"
          src="./public/img/package.png"
        />
        <div
          className="draggable-container"
          draggable={true}
          onClick={[Function]}
          onDragEnd={[Function]}
          onDragEnter={[Function]}
          onDragLeave={[Function]}
          onDragOver={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          style={Object {}}
        >
          New Folder
        </div>
      </li>
    </ul>
    <ul
      className="wonder-tree-node"
    >
      <li>
        <div
          className="item-triangle"
        />
        <img
          className="treeNode-icon"
          src="./public/img/package.png"
        />
        <div
          className="draggable-container"
          draggable={true}
          onClick={[Function]}
          onDragEnd={[Function]}
          onDragEnter={[Function]}
          onDragLeave={[Function]}
          onDragOver={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          style={Object {}}
        >
          New Folder 1
        </div>
      </li>
    </ul>
  </ul>
</article>
`;
